### PR TITLE
Local execution c example integration

### PR DESCRIPTION
**Pull request by aitorrf**
first version of local execution 
- native libraries integrated 
- pipeline execution
- temporal flavors solution  to create 2 different test apps
- jniLibs located inside of the library for this first version 
- assets models folder located in app

### CODE REVIEW

File: library/src/main/java/com/voysis/api/ITokenManager.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.voysis.api
+
+import com.voysis.model.request.Token
+
+interface ITokenManager {
```
Comment (lduffy-voysis) [inline library/src/main/java/com/voysis/api/ITokenManager.kt]: We've been using the convention of adding impl to interface implementation up until now so maybe remove the 'I' from ITokenManager to keep the convention consistent

Comment (oranoceallaigh) [reply library/src/main/java/com/voysis/api/ITokenManager.kt]: Oh $deity yes, none of this tagging class names with `I` or whatnot.

Even the `Impl` thing is somewhat questionable, it's a (lazy) style I follow but it isn't mandated by our code style so it's up to individuals if they want to go that way or not. As long as the name is expressive you should be good. I've not looked at what's implementing this interface in the PR but to throw out a random example, an acceptable set of names would be

```
interface TokenManager
class CloudTokenManager : TokenManager
class LocalTokenManager : TokenManager
```

File: app/src/main/java/com/voysis/android/MainActivity.kt [hunk 2]
```diff
@@ -73,8 +73,13 @@ class MainActivity : AppCompatActivity(), Callback {
     }
 
     private fun init() {
-        if (!::service.isInitialized)
-            service = ServiceProvider().makeLocal(applicationContext, config)
+        if (!::service.isInitialized) {
```
Comment (lduffy-voysis) [inline app/src/main/java/com/voysis/android/MainActivity.kt]: forgot to ask this before but why do we  have isInitialized here? is it to check that the models have loaded?

File: library/src/main/java/com/voysis/sevice/local/loader/ModelFileProvider.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.File
+import java.io.FileOutputStream
+
+/**
+ * Model file extract to get ready model files
+ */
+internal interface ModelFileProvider {
+
+    /**
+     * Read file and write it in an accessible localization
```
Comment (jcftang) [inline library/src/main/java/com/voysis/sevice/local/loader/ModelFileProvider.kt]: spelling mistake with location

File: library/src/main/java/com/voysis/sevice/local/loader/ModelFileProvider.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @param fName file Name to extract
+     * @return model extracted file localization
```
Comment (jcftang) [inline library/src/main/java/com/voysis/sevice/local/loader/ModelFileProvider.kt]: same spelling mistake

File: library/src/main/java/com/voysis/sevice/local/loader/ModelFileProvider.kt (hunk 3 above)
Comment (aitorrf) [reply library/src/main/java/com/voysis/sevice/local/loader/ModelFileProvider.kt]: good eye

File: library/src/main/java/com/voysis/sevice/local/observer/GenericObservable.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.voysis.sevice.local.observer
+
+import android.provider.SyncStateContract.Helpers.update
+
+open class GenericObservable {
+
+    val _observers = mutableListOf<GenericObserver>()
+
```
Comment (lduffy-voysis) [inline library/src/main/java/com/voysis/sevice/local/observer/GenericObservable.kt]: should this be private? Also, we don't use underscore notation in any of the libraries. Probably best to remove it.

**Pull request merged**